### PR TITLE
docs: add user guide for EnvoyProxy HPA

### LINKED ISSUE #2272: user guide for EnvoyProxy HPA
*Description*:
Add user guide for EnvoyProxy HPA

[optional *Relevant Links*:]
Related to #2257

### PR DESCRIPTION
**Pull request by ardikabs**
**What type of PR is this?**
Add user guide for EnvoyProxy HPA

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Related to #2257.
Close #2272

### CODE REVIEW

File: site/content/en/latest/user/customize-envoyproxy.md [hunk 1]
```diff
@@ -317,6 +317,36 @@ After applying the config, the bootstrap config will be overridden by the new co
 Any errors in the configuration will be surfaced as status within the `GatewayClass` resource.
 You can also validate this configuration using [egctl translate][].
 
+## Customize EnvoyProxy Horizontal Pod Autoscaler
+
+You can enable [Horizontal Pod Autoscaler](https://github.com/envoyproxy/gateway/issues/703) for EnvoyProxy Deployment via EnvoyProxy Config below:
```
Comment (Xunzhuo) [inline site/content/en/latest/user/customize-envoyproxy.md:322]: Should mention that we need a metrics-server installed in the cluster

Comment (ardikabs) [reply site/content/en/latest/user/customize-envoyproxy.md:322]: that's correct, let me update it

File: site/content/en/latest/user/customize-envoyproxy.md [hunk 2]
```diff
@@ -317,6 +317,38 @@ After applying the config, the bootstrap config will be overridden by the new co
 Any errors in the configuration will be surfaced as status within the `GatewayClass` resource.
 You can also validate this configuration using [egctl translate][].
 
+## Customize EnvoyProxy Horizontal Pod Autoscaler
+
+You can enable [Horizontal Pod Autoscaler](https://github.com/envoyproxy/gateway/issues/703) for EnvoyProxy Deployment. However, before enabling the HPA for EnvoyProxy, please ensure that the [metrics-server](https://github.com/kubernetes-sigs/metrics-server) component is installed in the cluster.
+
+Once confirmed, you apply it via EnvoyProxy Config as shown below:
```
Comment (shawnh2) [inline site/content/en/latest/user/customize-envoyproxy.md:324]:
```suggestion
Once confirmed, you can apply it via EnvoyProxy Config as shown below:
```

Comment (shawnh2) [inline site/content/en/latest/user/customize-envoyproxy.md:324]:
```suggestion
Once confirmed, you can apply it via EnvoyProxy Config as shown below:
```

**Review by shawnh2 — APPROVED** [review]
just a small nit. lgtm, thanks

**Review by zirain — APPROVED** [review]

**Pull request merged**